java/src/com/facebook/watchman/WatchmanClientImpl.java

```diff
@@ -255,11 +255,13 @@ public void call(Map<String, Object> message) throws Exception {
             .name(subscriptionId)
             .root(root)
             .build();
-        if (!subscriptions.containsKey(subscription)) {
+
+        Callback listener = subscriptions.get(subscription);
+        if (listener == null) {
```

Comment:
Hmm...wait, I take that back.  It looks like subscriptions is a ConcurrentHashMap, and those don't allow null keys or values.  Now I'm just confused...under what circumstance can this NPE?